src/main/java/core/basesyntax/strategy/impl/OperationStrategyImpl.java

```
@@ -0,0 +1,30 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation, OperationHandler>
+                                         operationHandlers) {
+        if (operationHandlers == null) {
+            throw new RuntimeException("Operation handlers map cannot be null.");
+        }
+        this.operationHandlers = Map.copyOf(operationHandlers);
```

The constructor checks the `operationHandlers` map for null and copies it with `Map.copyOf(...)`, but does not validate the map contents. The project guidance requires anticipating invalid inputs (including null parameters): "Remember that your code will need to be tested, so try to anticipate and handle all invalid input data in advance. For example: ... 3. Null parameters". Consider iterating the provided map and throwing a RuntimeException if any key or value is null to fail fast and provide a clearer error message when an invalid strategy map is supplied.